src/lwc/modal/__tests__/modal.test.js

```diff
@@ -0,0 +1,84 @@
+import { createElement } from "lwc";
+import Modal from "c/modal";
+
+const CSS_CLASS = "modal-hidden";
+
+describe("c-modal", () => {
+    let element;
+
+    afterEach(global.clearDOM);
+
+    beforeEach(() => {
+        element = createElement("c-modal", {
+            is: Modal,
+        });
+    });
+
+    it("renders the header content based on a public property and doesn not render the header slot", () => {
+        const HEADER = "The modal header";
+        element.header = HEADER;
+        document.body.appendChild(element);
+
+        return global.flushPromises().then(async () => {
+            // Query h2 element for header. We use a CSS selector to distinguish
+            // between the h2 tag for a header that's set via a public property
+            // vs a h2 tag that's set via the slot.
+            const headerEl = element.shadowRoot.querySelector("span");
+            expect(headerEl.textContent).toBe(HEADER);
+
+            const headerSlotEl = element.shadowRoot.querySelector('slot[name="header"]');
+            expect(headerSlotEl).toBeNull();
+            await expect(element).toBeAccessible();
+        });
+    });
+
+    // lwc-jest cannot validate slotchange events. We're only checking
+    // here if the empty h2 div is rendered.
+    it("renders the header slot when no public header property is set", () => {
+        document.body.appendChild(element);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return global.flushPromises().then(async () => {
+            // Query header slot element
+            const headerSlotEl = element.shadowRoot.querySelector('slot[name="header"]');
+            expect(headerSlotEl).not.toBeNull();
+
+            // TODO: Determine if we can require either the slot or header be populated?
```

Comment:
Are these todo's still open or are these intended to be resolved in another PR?

Reply:
No idea, I'm copying the modal from CaseMan (https://github.com/SalesforceFoundation/CaseMan/tree/master/force-app/main/default/lwc/modal) as Beth recommendation

Reply:
Fair enough. @bethbrains Any concerns if we delete the todo from this component or is it worth keeping as a reminder that something eventually needs to be done?
Also @dcaballeroSf Can you add a comment block to the top of the JS file for the parent modal component to indicate that it was copied from CaseMan just so we remember in the future?

Reply:
yeah, it should be fine to remove the TODO. @jjbennett likely wrote this test.